File: sdks/typescript/examples/basic.ts
/**
 * Basic Example: Create a session with AI Gate
 */

import { AIGateClient, AIGateError } from '../src';

async function main() {
  // Initialize the client with your project ID
  const client = new AIGateClient({
    projectId: process.env.AI_GATE_PROJECT_ID || 'your-project-id',
    debug: true
  });

  try {
    console.log('🚀 Creating OpenAI Realtime session...\n');

    // Create a session
    const session = await client.createSession({
      model: 'gpt-4o-realtime-preview-2024-10-01',
      voice: 'echo',
      temperature: 0.7,
      instructions: 'You are a helpful and friendly assistant.'
    });

    console.log('✅ Session created successfully!\n');
    console.log('📋 Session Details:');
    console.log(`  - Session ID: ${session.session_id}`);
    console.log(`  - OpenAI Session: ${session.id}`);
    console.log(`  - Model: ${session.metadata.model}`);
    console.log(`  - Voice: ${session.metadata.voice}`);
    console.log(`  - TURN Server: ${session.metadata.turn_server}`);
    console.log(`  - Expires: ${new Date(session.expires_at * 1000).toLocaleString()}`);
    console.log(`  - Created: ${session.metadata.created_at}\n`);

    console.log('🔐 Client Secret (for WebRTC):');
    console.log(`  ${session.client_secret?.value.substring(0, 20)}...`);
    console.log(`  Expires: ${new Date((session.client_secret?.expires_at || 0) * 1000).toLocaleString()}\n`);

    console.log('🌐 TURN Credentials:');
    session.turn_credentials.iceServers.forEach((server, i) => {
      console.log(`  Server ${i + 1}:`);
      console.log(`    URLs: ${Array.isArray(server.urls) ? server.urls.join(', ') : server.urls}`);
      if (server.username) {
        console.log(`    Username: ${server.username}`);
      }
    });

  } catch (error) {
    if (error instanceof AIGateError) {
      console.error('❌ AI Gate Error:', error.message);
      console.error('   Status Code:', error.statusCode);

      if (error.statusCode === 402) {
        console.error('   💡 Your quota is exceeded. Please upgrade your plan.');
        console.error(`   Upgrade URL: ${(error.response as any)?.upgrade_url}`);
      } else if (error.statusCode === 429) {
        console.error('   💡 Rate limit exceeded. Please try again later.');
      }
    } else {
      console.error('❌ Unexpected error:', error);
    }
    process.exit(1);
  }
}

main();

File: sdks/python/examples/basic.py
"""
Basic Example: Create a session with AI Gate
"""

import os
from ai_gate import AIGateClient, AIGateError


def main():
    # Initialize the client with your project ID
    client = AIGateClient(
        project_id=os.getenv("AI_GATE_PROJECT_ID", "your-project-id"),
        debug=True
    )

    try:
        print("🚀 Creating OpenAI Realtime session...\n")

        # Create a session
        session = client.create_session(
            model="gpt-4o-realtime-preview-2024-10-01",
            voice="echo",
            temperature=0.7,
            instructions="You are a helpful and friendly assistant."
        )

        print("✅ Session created successfully!\n")
        print("📋 Session Details:")
        print(f"  - Session ID: {session['session_id']}")
        print(f"  - OpenAI Session: {session['id']}")
        print(f"  - Model: {session['metadata']['model']}")
        print(f"  - Voice: {session['metadata']['voice']}")
        print(f"  - TURN Server: {session['metadata']['turn_server']}")
        print(f"  - Created: {session['metadata']['created_at']}\n")

        if "client_secret" in session:
            print("🔐 Client Secret (for WebRTC):")
            print(f"  {session['client_secret']['value'][:20]}...")
            print()

        print("🌐 TURN Credentials:")
        for i, server in enumerate(session["turn_credentials"]["iceServers"]):
            print(f"  Server {i + 1}:")
            urls = server["urls"]
            if isinstance(urls, list):
                urls = ", ".join(urls)
            print(f"    URLs: {urls}")
            if "username" in server:
                print(f"    Username: {server['username']}")
        print()

    except AIGateError as error:
        print(f"❌ AI Gate Error: {error}")
        print(f"   Status Code: {error.status_code}")

        if error.status_code == 402:
            print("   💡 Your quota is exceeded. Please upgrade your plan.")
            if error.response:
                print(f"   Upgrade URL: {error.response.get('upgrade_url')}")

        elif error.status_code == 429:
            print("   💡 Rate limit exceeded. Please try again later.")

    except Exception as error:
        print(f"❌ Unexpected error: {error}")

    finally:
        client.close()


if __name__ == "__main__":
    main()
